### ISSUE TITLE
Plugin doesn't seem to be working

### ISSUE THREAD

**Issue by DevJake** [lumoe/obsidian-rollover-daily-todos#43]
Hey there,

This plugin looks great! I installed it earlier and I've spent a fair while fiddling with the settings, text layouts, etc. but to no avail. I can't seem to get it to work at all... The settings are definitely correct, the naming of daily notes is correct, and even the structure of tasks is correct (I checked each for matches to the stated RegEx query, and all of the incomplete ones matched perfectly). So I'm not sure what the issue could be, if it's my side or something with the plugin. 

If it means anything, I've recently begun using the _Live Preview_ mode for Obsidian, which I've noticed has caused some other plugins to break, so that could be a potential? 

Let me know if there's any more specifics or information you need!! 😄

**Comment by LarsKumbier** [comment]
Hey, the project seems to be dead. I've forked it in https://github.com/LarsKumbier/obsidian-rollover-daily-todos - I've added the option rollover content blocks as well (see #41 ), if you want to check it out.

As for your bug: the code works fine for me - did you activate the "daily notes" plugin or do you use a different one?

Does the console give any output (accessible via CTRL+SHIFT+i)?

**Comment by DevJake** [comment]
Hey Lars,

Yes, a dead project indeed :( 

I have the Daily Notes plugin active, and I've just tried reinstalling this one, with no luck. I don't use a different plugin for daily notes. 

I'll check out your repo. Hopefully you're able to keep this plugin running!!

**Comment by LarsKumbier** [comment]
Did you check the settings? I'm unsure if it will work, if you do not give a headline to put the todos under, haven't tried it. 
Did you get any output / errors in the developer console?

**Comment by DevJake** [comment]
Yep, the settings should all be correct. My Daily Notes use the following template:

````markdown
---
---

#notes/daily

# Notes

# Tasks
## Outstanding
```tasks
path includes Daily Notes/{{date:DD-MM-YYYY}}
not done
hide backlink
```

## Complete
```tasks
path includes Daily Notes/{{date:DD-MM-YYYY}}
done
hide backlink
```

## All
- [ ] #task ...
````

The `tasks` code block is from the [obsidian-tasks plugin](https://github.com/schemar/obsidian-tasks). I've configured the rollover to move all tasks to the `## All` header, at the bottom. Here's a screenshot of my configuration for rollover:
![Rollover Daily Todos configuration](https://user-images.githubusercontent.com/4295059/148268086-28076440-b5cb-4a54-bb32-ee9ed679c90c.png)

As you see, it's set to use the `## All` heading. When pressing `ctrl+P` and executing the command `Rollover Daily Todos: Rollover Todos Now`, nothing happens in the editing window, and the console (`ctrl+shift+i`) remains empty of any sort of warnings, errors, or general messages, even with all message types (including `Verbose`) turned on. 

Hope this helps a bit. There's a potential it could be colliding with one of the other plugins I have installed, but I really doubt it as the only ones that affect daily notes are the two core plugins for daily notes and templates. 😕

**Comment by LarsKumbier** [comment]
It's strange that you do not see *any* output in the console window - there should be at least *some* startup information... Could you install my version instead and activate the debug setting I just added? There should be some output in the console info from the plugin.

**Comment by DevJake** [comment]
Apologies for the delay, had to quickly learn how to build yours. I'm now using it, and I get this message in the console:

```
Failed to load resource: 
github.com/shichongr.../1.0.5/styles.css:1

the server responded with a status of 404 ()
```

Followed soon by `Rollover Daily Todos: styles.css not found`

With a bit more fiddling and running the aforementioned `Rollover` command, I managed to get this:
![image](https://user-images.githubusercontent.com/4295059/148284837-1340ecf5-4b6c-431e-a4d0-811d306c47d3.png)
Apologies for it being a screenshot, it was just simpler than copying and fixing formatting 😅

I did also try coping in a `styles.css` from another plugin, but whether it has content or I remove it all, neither works.

**Comment by DevJake** [comment]
This is the output with Debug mode enabled:
![image](https://user-images.githubusercontent.com/4295059/148285816-f41d87d0-f88d-4326-929a-50528812780c.png)

And the relevant settings for your version:
![image](https://user-images.githubusercontent.com/4295059/148285857-6dbb115b-7e34-4a11-b8a6-67774e4083e3.png)

**Comment by LarsKumbier** [comment]
I'm still finding my way around the source code, might have found a missing check. Could you pull again and retry?

**Comment by DevJake** [comment]
> This is the output with Debug mode enabled: ![image](https://user-images.githubusercontent.com/4295059/148286196-7e2b711e-b701-41e3-a7c1-1f30e29a0790.png)
> 
> And the relevant settings for your version: ![image](https://user-images.githubusercontent.com/4295059/148285857-6dbb115b-7e34-4a11-b8a6-67774e4083e3.png)

Of course. I pulled, re-built and ran the same commands with the same setup as the quoted code, but still got the same error in the console. 

It's an odd error because - to me at least - it's so vague and non-descriptive. My guess would be with me using the new Live Preview editor over the legacy editor, which has caused several other plugins to either partially or completely break. I'll switch back to legacy, refresh and see if it changes anything...

**Comment by LarsKumbier** [comment]
It's odd that the line numbers did not change - could double-check that you are using my fork and that you uninstalled the original plugin? You can also start adding some console.logs by yourself in the code - the basename is only used half a dozen times.

**Comment by DevJake** [comment]
I deleted the entire plugin directory, then used `git clone` on your repo. `cd` into the directory, run `npm i` then `npm run build`. Refresh the plugins list in Obsidian, enable yours, configure yours as shown prior, then repeat the `Rollover` command from the command palette menu. I still receive the same error messages as before, same erroneous lines too. 

I'll have a look at adding some `console.logs` too, see what I can get out of it. Thanks for looking into this Lars 😀

**Comment by sczimmerman** [comment]
tldr; It looks like the plugin doesn't rollover todos from one month to another.

I'm experiencing the same issue with this plugin, and I noticed that your original post was created on January 1. I began using this plugin in January, and first noticed yesterday, February 1st, that my tasks were no longer rolling over. On a hunch, I tried manually copying my unchecked items from 1/31 to my 2/1 note. Today, 2/2, my tasks rolled over to today's daily note.

**Comment by phirework** [comment]
FWIW - I started having issues with it recently when I enabled the new Live Preview editor. I disabled the "delete rolled over tasks" option, and now manually run the "rollover" task when a new Daily note is created, and that works fine. It's not ideal, but it's better than copy/pasting.

**Comment by bchmura** [comment]
Like phirework, this only works for me when I manually invoke it.  The automatic produces no errors in the console.  On Obsidian 1.0.3 and latest plugin version (also win 10)

**Comment by tim-hilt** [comment]
@lumoe I think this one can be closed
